packages/main/src/plugin/tasks/task-manager.ts

```diff
@@ -113,11 +113,23 @@ export class TaskManager {
     this.setStatusBarEntry(true);
   }
 
-  public createTask(options?: { title?: string; action?: TaskAction }): Task {
+  public createTask(options?: {
+    title?: string;
+    action?: TaskAction;
+    cancellable?: boolean;
+    cancellationTokenSourceId?: number;
+  }): Task {
     this.taskId++;
 
     const task = new TaskImpl(`task-${this.taskId}`, options?.title ? options.title : `Task ${this.taskId}`);
     task.action = options?.action;
+    task.cancellable = options?.cancellable ?? false;
+    if (task.cancellable && !options?.cancellationTokenSourceId) {
+      throw new Error('cancellable task requires a cancellationTokenSourceId');
+    }
```

Comment:
Is the other way possible, to have a `cancellationTokenSourceId` but not `cancellable`?

Reply:
no

Reply:
in this case, IMO you would have to add the check, or not ask for `cancellable` from the consumer and have `cancellable = !! cancellationTokenSourceId`

Reply:
updated